File: tkpdabstraction/src/main/java/com/tokopedia/abstraction/di/module/AppModule.java
package com.tokopedia.abstraction.di.module;

import android.content.Context;

import com.tokopedia.abstraction.di.qualifier.ApplicationContext;
import com.tokopedia.abstraction.di.scope.ApplicationScope;

import dagger.Module;
import dagger.Provides;


/**
 * @author kulomady on 1/9/17.
 */
@Module
public class AppModule {

    private final Context context;

    public AppModule(Context context) {
        this.context = context;
    }

    @ApplicationScope
    @Provides
    @ApplicationContext
    public Context provideContext() {
        return context.getApplicationContext();
    }

}

File: libraries/promotions_target/src/main/java/com/tokopedia/promotionstarget/di/components/AppModule.kt
package com.tokopedia.promotionstarget.di.components

import android.content.Context
import com.tokopedia.promotionstarget.di.scopes.PromoTargetScope
import dagger.Module
import dagger.Provides

@PromoTargetScope
@Module
class AppModule(val applicationContext: Context) {

    @Provides
    @PromoTargetScope
    fun provideContext() = applicationContext
}